File: Programming-for-QA-sep2023/08.UnitTesting-Methods/TestApp.UnitTests/FactorialTests.cs
using NUnit.Framework;
using System;

namespace TestApp.UnitTests;

public class FactorialTests
{
    [Test]
    public void CalculateFactorial_InputZero_ReturnsOne()
    {
        //Arrange
        int n = 0;

        //Act
        int result = Factorial.CalculateFactorial(n);

        //Assert
        Assert.AreEqual(1, result);
    }

    [Test]
    public void CalculateFactorial_InputPositiveNumber_ReturnsCorrectFactorial()
    {
        //Arrange
        int n = 3;

        //Act
        int result = Factorial.CalculateFactorial(n);

        //Assert
        Assert.AreEqual(6, result);
    }

    [Test]
    public void CalculateFactorial_InputNegativeNumber_ThrowsException()
    {
        //Arrange
        int n = -1;

        //Act + Assert
        Assert.Throws<ArgumentOutOfRangeException>(() => Factorial.CalculateFactorial(n));
    }

    [Test, Ignore("not implemented")]
    [Category("Ignored UnitTests")]
    public void TestTest()
    {
        
    }
}

File: QA_FrontEnd-Test-Automation-May2024/04-QUnit-Tests/test/factorial-tests.js
const {factorial} = require('../test_functions.js');

QUnit.module('test factorial function', () => {
    QUnit.test('pass 5 as parameter, get 120 as a result', (assert) => {
        assert.equal(factorial(5), 120, 'f(5) should return 120')
    })

    QUnit.test('pass 0 as parameter, get 1 as a result', (assert) => {
        assert.equal(factorial(5), 120, 'f(0) should return 1')
    })

    QUnit.test('pass -1 as parameter, get 1 as a result', (assert) => {
        assert.equal(factorial(5), 120, 'f(-1) should return 1')
    })
})